internal/kafka/internal/services/kafka.go

```diff
@@ -235,7 +230,12 @@ func (k *kafkaService) RegisterKafkaJob(kafkaRequest *dbapi.KafkaRequest) *error
 	// we need to pre-populate the ID to be able to reserve the quota
 	kafkaRequest.ID = api.NewID()
 
-	hasCapacity, err := k.HasAvailableCapacityInRegion(kafkaRequest)
+	kafkaInstanceSize, e := k.kafkaConfig.GetKafkaInstanceSize(kafkaRequest.InstanceType, kafkaRequest.SizeId)
+	if e != nil {
+		return errors.NewWithCause(errors.ErrorGeneral, e, fmt.Sprintf("Failed to check kafka capacity for region '%s' and instance type '%s'", kafkaRequest.Region, kafkaRequest.InstanceType))
+	}
+
+	hasCapacity, err := k.HasAvailableCapacityInRegion(kafkaRequest, kafkaInstanceSize.CapacityConsumed)
```

Comment:
No need to add this parameter here: `HasAvailableCapacityInRegion` can retrieve the `KafkaInstanceSize` with `k.kafkaConfig.GetKafkaInstanceSize(kafkaRequest.InstanceType, kafkaRequest.SizeId)`

Reply:
> There a re a number of methods that receives both a `KafkaRequest,KafkaConfig` pair and a `requiredSize` parameter. I believe the latter could be inferred from the former:
> 
> ```go
> kafkaConfig.GetKafkaInstanceSize(kafkaRequest.InstanceType, kafkaRequest.SizeId)
> ```

yes. this requires a lot of refactoring to the quota related stuff